src/lightning_app/api/http_methods.py

```diff
@@ -29,61 +29,82 @@ def __init__(self, route: str, method: Callable, method_name: Optional[str] = No
             timeout: The time in seconds taken before raising a timeout exception.
         """
         self.route = route
-        self.component_name = method.__self__.name
+        self.attached_to_flow = hasattr(method, "__self__")
         self.method_name = method_name or method.__name__
         self.method_annotations = method.__annotations__
         # TODO: Validate the signature contains only pydantic models.
         self.method_signature = inspect.signature(method)
+        if not self.attached_to_flow:
+            self.component_name = method.__name__
+            self.method = method
+        else:
+            self.component_name = method.__self__.name
+
         self.timeout = timeout
         self.kwargs = kwargs
 
     def add_route(self, app: FastAPI, request_queue: Queue, responses_store: Dict[str, Any]) -> None:
-        # 1: Create a proxy function with the signature of the wrapped method.
-        fn = deepcopy(_signature_proxy_function)
-        fn.__annotations__ = self.method_annotations
-        fn.__name__ = self.method_name
-        setattr(fn, "__signature__", self.method_signature)
-
-        # 2: Get the route associated with the http method.
+        # 1: Get the route associated with the http method.
         route = getattr(app, self.__class__.__name__.lower())
 
-        request_cls = CommandRequest if self.route.startswith("/command/") else APIRequest
-
-        # 3: Define the request handler.
-        @wraps(_signature_proxy_function)
-        async def _handle_request(*args, **kwargs):
-            async def fn(*args, **kwargs):
-                request_id = str(uuid4()).split("-")[0]
-                logger.debug(f"Processing request {request_id} for route: {self.route}")
-                request_queue.put(
-                    request_cls(
-                        name=self.component_name,
-                        method_name=self.method_name,
-                        args=args,
-                        kwargs=kwargs,
-                        id=request_id,
+        if not self.attached_to_flow:
+            fn = deepcopy(_signature_proxy_function)
+            fn.__annotations__ = self.method_annotations
+            fn.__name__ = self.method_name
+            setattr(fn, "__signature__", self.method_signature)
+
+            request_cls = CommandRequest if self.route.startswith("/command/") else APIRequest
+
+            # 3: Define the request handler.
```

Comment:
The comments are not in the correct order.

Reply:
Done !